packages/jest-fake-timers/src/jestFakeTimers.ts

```diff
@@ -374,8 +374,8 @@ export default class FakeTimers<TimerRef> {
   private _fakeClearTimer(timerRef: TimerRef) {
     const uuid = this._timerConfig.refToId(timerRef);
 
-    if (uuid && this._timers.hasOwnProperty(uuid)) {
-      delete this._timers[String(uuid)];
+    if (uuid && this._timers.has(uuid)) {
```

Comment:
do we need the `has` check? I'd guess `delete` just no-ops if the key is not there